lgc/patch/PatchLoopMetadata.cpp

```diff
@@ -24,91 +24,86 @@
+  GfxIpVersion m_gfxIp;
+};
 
-// =====================================================================================================================
-// Initializes static members.
-char SpirvLowerLoopUnrollControl::ID = 0;
+} // anonymous namespace
 
 // =====================================================================================================================
-// Pass creator, creates the pass of SPIR-V lowering operations for loop unroll control
-//
-// @param forceLoopUnrollCount : Force loop unroll count
-ModulePass *createSpirvLowerLoopUnrollControl(unsigned forceLoopUnrollCount) {
-  auto pass = new SpirvLowerLoopUnrollControl(forceLoopUnrollCount);
-  return pass;
-}
+// Initializes static members.
+char PatchLoopMetadata::ID = 0;
 
 // =====================================================================================================================
-SpirvLowerLoopUnrollControl::SpirvLowerLoopUnrollControl()
-    : SpirvLower(ID), m_forceLoopUnrollCount(0), m_disableLicm(false), m_disableLoopUnroll(false) {
+// Pass creator, creates the pass for patching loop metadata
+ModulePass *lgc::createPatchLoopMetadata() {
+  return new PatchLoopMetadata();
 }
 
 // =====================================================================================================================
-//
-// @param forceLoopUnrollCount : Force loop unroll count
-SpirvLowerLoopUnrollControl::SpirvLowerLoopUnrollControl(unsigned forceLoopUnrollCount)
-    : SpirvLower(ID), m_forceLoopUnrollCount(forceLoopUnrollCount), m_disableLicm(false), m_disableLoopUnroll(false) {
+PatchLoopMetadata::PatchLoopMetadata()
+    : Patch(ID), m_forceLoopUnrollCount(0), m_disableLicm(false), m_disableLoopUnroll(false) {
 }
 
 // =====================================================================================================================
-// Executes this SPIR-V lowering pass on the specified LLVM module.
+// Executes this LLVM patching pass on the specified LLVM module.
 //
 // @param [in,out] module : LLVM module to be run on
-bool SpirvLowerLoopUnrollControl::runOnModule(Module &module) {
-  LLVM_DEBUG(dbgs() << "Run the pass Spirv-Lower-Unroll-Control\n");
-
-  SpirvLower::init(&module);
-
-  if (m_context->getPipelineContext()) {
-    auto shaderOptions = &(m_context->getPipelineShaderInfo(m_shaderStage)->options);
-    if (shaderOptions->forceLoopUnrollCount > 0)
+bool PatchLoopMetadata::runOnModule(Module &module) {
+  LLVM_DEBUG(dbgs() << "Run the pass patch-loop-metadata\n");
+
+  Patch::init(&module);
+  PipelineState *m_pipelineState = getAnalysis<PipelineStateWrapper>().getPipelineState(&module);
+  m_gfxIp = m_pipelineState->getTargetInfo().getGfxIpVersion();
+  bool Changed = false;
+
+  for (Function &func : module) {
+    ShaderStage stage = getShaderStage(&func);
+    if (stage == -1)
+      continue;
+    auto shaderOptions = &m_pipelineState->getShaderOptions(stage);
+    if (shaderOptions) {
```

Comment:
nit: you can fold these two and put the definition in the if condition if it's not used elsewhere, i.e.:
` if (auto shaderOptions = &...)`

Reply:
Done